src/System.Device.Gpio/System/Device/Pwm/Drivers/Windows10PwmDriverChannel.Windows.cs

```diff
@@ -26,8 +26,8 @@ public void ChangeDutyCycle(double dutyCyclePercentage)
 
         public void Start(double dutyCyclePercentage)
         {
-            this.ChangeDutyCycle(dutyCyclePercentage);
             _winPin?.Start();
```

Comment:
I suppose there is no way of calling the WinRT Api Start and pass in the duty cycle as well in order to address Jon's concern?

Reply:
Unfortunately the Start() API currently has no way to pass in the duty cycle.
The default duty cycle is 0%. Perhaps setting the duty cycle before and after Start() would address the concern

Reply:
Muhamad Lotfy wrote the WinRT API for PWM. It does seem like you should set duty cycle before calling Start(), but that's how I had it and you said it doesn't work unless you call it after Start()? Since the default is zero, it doesn't seem like it would hurt to have a small amount of time of 0%.

---
In reply to: [241921911](https://github.com/dotnet/iot/pull/97#discussion_r241921911) [](ancestors = 241921911)

Reply:
I'm fine with setting duty cycle before and after calling start, but we should add a comment on the second operation to set duty cycle (after calling start) and link an issue saying that this should be removed once this is fixed in the WinRT side as it seems that it was supposed to work if you set duty cycle before calling start according to @JohnTasler 's comment.

Reply:
I've created the following issue and added a comment. #109

Reply:
Shouldn't the issue be against the Low Level device driver in Windows?

---
In reply to: [243551826](https://github.com/dotnet/iot/pull/97#discussion_r243551826) [](ancestors = 243551826)

Reply:
First discuss with Jordan. If it's an actual bug in the WinRT API (and not just a usage bug), then write a new OS bug, and possibly refer to its internal bug id in the GitHub issue.